idaes/power_generation/carbon_capture/mea_solvent_system/unit_models/column.py

```diff
@@ -485,66 +472,268 @@ def _make_performance(self):
-                                      components at the interface ''')
-        self.N_v = Var(self.t,
-                       self.liquid_phase.length_domain,
-                       dcomp,
-                       domain=Reals,
-                       initialize=0.0,
-                       units=pyunits.mol / (pyunits.s * pyunits.m),
-                       doc='''Moles of diffusing species transfered
-                                     into liquid ''')
-        self.enhancement_factor = Var(self.t,
+        
+        # Liquid holdup model    
+        if self.config.fix_column_pressure or self.config.fix_column_pressure_drop:
+            # reference: Tsai correlation,regressed by Chinen et al. 2018
+            def rule_holdup_liq(blk, t, x):
+                return (blk.holdup_parA * (
+                    (blk.velocity_liq[t, x]*blk.area_column/\
+                     (CONST.pi*blk.dh_ref*(blk.S_ref)**(2)*(g_ref)**(2/3))) *
+                    (blk.liquid_phase.properties[t, x].visc_d_phase['Liq'] /
+                     (blk.liquid_phase.properties[t, x].mw/\
+                        blk.liquid_phase.properties[t, x].vol_mol_phase['Liq']))**(0.333)
+                    )**blk.holdup_parB)
+    
+            self.holdup_liq = Expression(self.flowsheet().time,
+                                         self.liquid_phase.length_domain,
+                                         rule=rule_holdup_liq,
+                                         doc='Volumetric liquid holdup [-]')
+        else:
+            # Reference: Stichlmair correlation 
+            def rule_holdup_liq(blk, t, x):
+                x_t = blk.liquid_phase.length_domain.at(blk.zi[x].value - 1)
+                Fr_l = ((blk.velocity_liq[t, x])**2)*blk.a_ref/(g_ref*(blk.eps_ref)**4.65)
+                h0 = 0.555*(Fr_l)**(1/3)
+                return h0*(1 + 20*(blk.config.vapor_side.property_package.deltaP[t, 1-x]*(x - x_t)/\
+                            ((blk.liquid_phase.properties[t, x].mw/\
+                        blk.liquid_phase.properties[t, x].vol_mol_phase['Liq'])*g_ref)**2))
+    
+            self.holdup_liq = Expression(self.flowsheet().time,
+                                         self.liquid_phase.length_domain,
+                                         rule=rule_holdup_liq,
+                                         doc='Volumetric liquid holdup [-]')
+
+        # Vapor holdup model
+        # reference: Tsai correlation,regressed by Chinen et al. 2018
+        def rule_holdup_vap(blk, t, x):
+            return blk.eps_ref - blk.holdup_liq[t, x]
+
+        self.holdup_vap = Expression(self.flowsheet().time,
+                                     self.vapor_phase.length_domain,
+                                     rule=rule_holdup_vap,
+                                     doc='Volumetric vapor holdup [-]')
+        
+        # Column pressure profile
+        if self.config.fix_column_pressure:
+            self.column_pressure = Param(
```

Comment:
This should not be a Param. Users should either provide a model for delta P, or just fix the variable directly.

On that note, we should be creating a reference to one of the delta P variables before this, along with a constraint equating delta P in each phase. Then, users can just work with the unit level delta P without needing to be concerned about the control volumes.